src/tss2-fapi/ifapi_config.c

```diff
@@ -153,6 +153,49 @@ ifapi_config_initialize_async(IFAPI_IO *io)
     return TSS2_RC_SUCCESS;
 }
 
+/**
+ * Expand user symbol in path.
+ *
+ * "~" and "$HOME" will be replaces by the value of the HOME environment
+ * variable.
+ *
+ * @param[in, out] The path.
+ * @retval TSS2_RC_SUCCESS on success
+ * @retval TSS2_FAPI_RC_MEMORY if not enough memory can be allocated.
+ */
+
+static TSS2_RC
+expand_home(char **path) {
+    size_t startPos = 0;
+    TSS2_RC r;
+
+    /* Check whether usage of home directory in pathname */
+    if (strncmp("~", *path, 1) == 0) {
+        startPos = 1;
+    } else if (strncmp("$HOME", *path, 5) == 0) {
+        startPos = 5;
+    }
+
+    /* Replace home abbreviation in path. */
+    char *newPath = NULL;
+    if (startPos != 0) {
+        LOG_DEBUG("Expanding path %s to user's home", *path);
+        char *homeDir = getenv("HOME");
+        return_if_null(homeDir, "Home directory can't be determined.",
+                       TSS2_FAPI_RC_BAD_PATH);
+        if (strncmp(&(*path)[startPos], IFAPI_FILE_DELIM, strlen(IFAPI_FILE_DELIM)) == 0) {
+            startPos += strlen(IFAPI_FILE_DELIM);
+        }
+        r = ifapi_asprintf(&newPath, "%s%s%s", homeDir, IFAPI_FILE_DELIM,
+                           &(*path)[startPos]);
+        return_if_error(r, "Out of memory.");
+
+        SAFE_FREE(*path);
+        *path = newPath;
+    }
+    return TSS2_RC_SUCCESS;
+}
```

Comment:
This only expands the first portion, which is likely to cover the majority of use cases. Perhaps something in the future using wordexp.h, but thats a GNUC thing IIUC.

Reply:
Tank you for the review. I have fixed your first two points and the CI error. Should I change anything related to your point three?

Reply:
More of an FYI, depends on how feature rich you want this to be, but this seems fine.